presto-geospatial/src/main/java/com/facebook/presto/geospatial/GeoFunctions.java

```diff
@@ -0,0 +1,513 @@
+    {
+        OGCGeometry leftGeometry = deserialize(left);
+        OGCGeometry rightGeometry = deserialize(right);
+        OperatorSimpleRelation intersectsOperator = OperatorIntersects.local();
+        return intersectsOperator.execute(leftGeometry.getEsriGeometry(), rightGeometry.getEsriGeometry(), leftGeometry.getEsriSpatialReference(), null);
+    }
+
+    @SqlNullable
+    @Description("Returns true if and only if left geometry overlaps right geometry")
+    @ScalarFunction("st_overlaps")
+    @SqlType(StandardTypes.BOOLEAN)
+    public static Boolean stOverlaps(@SqlType(GEOMETRY) Slice left, @SqlType(GEOMETRY) Slice right)
+    {
+        OGCGeometry leftGeometry = deserialize(left);
+        OGCGeometry rightGeometry = deserialize(right);
+        OperatorSimpleRelation overlapsOperator = OperatorOverlaps.local();
+        return overlapsOperator.execute(leftGeometry.getEsriGeometry(), rightGeometry.getEsriGeometry(), leftGeometry.getEsriSpatialReference(), null);
+    }
+
+    @SqlNullable
+    @Description("Returns true if and only if left geometry has the specified DE-9IM relationship with right geometry")
+    @ScalarFunction("st_relate")
+    @SqlType(StandardTypes.BOOLEAN)
+    public static Boolean stRelate(@SqlType(GEOMETRY) Slice left, @SqlType(GEOMETRY) Slice right, @SqlType(StandardTypes.VARCHAR) Slice relation)
+    {
+        OGCGeometry leftGeometry = deserialize(left);
+        OGCGeometry rightGeometry = deserialize(right);
+        return leftGeometry.relate(rightGeometry, relation.toStringUtf8());
+    }
+
+    @SqlNullable
+    @Description("Returns true if and only if left geometry touches right geometry")
+    @ScalarFunction
+    @SqlType(StandardTypes.BOOLEAN)
+    public static Boolean stTouches(@SqlType(GEOMETRY) Slice left, @SqlType(GEOMETRY) Slice right)
+    {
+        OGCGeometry leftGeometry = deserialize(left);
+        OGCGeometry rightGeometry = deserialize(right);
+        OperatorSimpleRelation touchesOperator = OperatorTouches.local();
+        return touchesOperator.execute(leftGeometry.getEsriGeometry(), rightGeometry.getEsriGeometry(), leftGeometry.getEsriSpatialReference(), null);
+    }
+
+    @SqlNullable
+    @Description("Returns true if and only if left geometry is within right geometry")
+    @ScalarFunction("st_within")
+    @SqlType(StandardTypes.BOOLEAN)
+    public static Boolean stWithin(@SqlType(GEOMETRY) Slice left, @SqlType(GEOMETRY) Slice right)
+    {
+        OGCGeometry leftGeometry = deserialize(left);
+        OGCGeometry rightGeometry = deserialize(right);
+        OperatorSimpleRelation withinOperator = OperatorWithin.local();
+        return withinOperator.execute(leftGeometry.getEsriGeometry(), rightGeometry.getEsriGeometry(), leftGeometry.getEsriSpatialReference(), null);
+    }
+
+    private static OGCGeometry getGeometry(OGCGeometry geometry, String function, String...validTypes)
```

Comment:
nit: consider adding requireNotNull(geometry) check in the beginning of this method.